samples/nextjs/src/lib/with-personalization-and-tracking.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  isServer,
+  LayoutPersonalizationService,
+  withSitecorePersonalizationContext,
+  areQueryParamsReady,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { TrackingService } from '../../../../packages/sitecore-jss-tracking/types';
+import { SitecorePageProps } from 'lib/page-props';
+import { useRouter } from 'next/router';
+
+export function withCurrentPageTracking<T extends SitecorePageProps>(
```

Comment:
This signature doesn't fit with our HOC's signature.
Use: `withNotFoundPageTracking(options)(Component)`